### PR TITLE
1953 bom delete schema change

### PR DESCRIPTION
**Pull request by superhvarn**
## Changes

Added date deleted to unit

## Checklist

It can be helpful to check the `Checks` and `Files changed` tabs.
Please review the [contributor guide](https://nerdocs.atlassian.net/wiki/spaces/NER/pages/8060929/Software+Contributor+Guide) and reach out to your Tech Lead if anything is unclear.
Please request reviewers and ping on slack only after you've gone through this whole checklist.

- [x] All commits are tagged with the ticket number
- [x] No linting errors / newline at end of file warnings
- [x] All code follows repository-configured prettier formatting
- [x] No merge conflicts
- [x] All checks passing
- [x] Screenshots of UI changes (see Screenshots section)
- [x] Remove any non-applicable sections of this template
- [x] Assign the PR to yourself
- [x] No `yarn.lock` changes (unless dependencies have changed)
- [x] Request reviewers & ping on Slack
- [x] PR is linked to the ticket (fill in the closes line below)

Closes # (issue #1953)

### CODE REVIEW

File: src/backend/tests/test-data/projects.test-data.ts [hunk 1]
```diff
@@ -162,7 +162,8 @@ export const prismaMaterialType: PrismaMaterialType = {
 };
 
 export const prismaUnit: Unit = {
-  name: 'FT'
+  name: 'FT',
+  dateDeleted: null
```
Comment (Aaryan1203) [inline src/backend/tests/test-data/projects.test-data.ts:166]: Is this necessary if the date deleted field is optional?

Comment (superhvarn) [reply src/backend/tests/test-data/projects.test-data.ts:166]: It required me to do it. Otherwise, the build failed.

**Review by Aaryan1203 — APPROVED** [review]
LGTM!

**Review by RChandler234 — CHANGES_REQUESTED** [review]
Ok so looking through the schema again, I think we'll actually need to hard delete units, manufacturers and material types. Because we're using their name as the id, if we mark them as deleted, you won't be able to make a unit (or the other two) of that type again which isn't good. I'm gonna close this PR/ ticket and get you a new one

**Comment by RChandler234** [comment]
Not doing

**Pull request closed without merging**